### ISSUE TITLE
Improve Contributor Experience

### ISSUE THREAD

**Issue by TheRoyalTnetennba** [GoogleCloudPlatform/ruby-docs-samples#396]
- [ ] Kokoro feedback should be fast and reliable
- [ ] Kokoro should test on OSx
- [ ] Instructions on how to run the tests locally should be visible and simple
- [ ] Running tests in parallel should create no problems
- [ ] There should be linter rules and Kokoro should check them
- [ ] README generation should be automated where reasonable.

**Comment by tbpg** [comment]
What is README generation? Do we need READMEs?

**Comment by frankyn** [comment]
Hi @tbpg,

Sample tests may have different setup instructions. Additionally, we've in the past added CLI commands to our README's and usually a manual process. It would be nice to have automation for these tasks similar to Python-docs-samples.

@TheRoyalTnetennba It might be a good idea to flesh out this work item more.
